src/age_types.rs

```diff
@@ -119,6 +139,60 @@ where
     }
 }
 
+/// Represents path in graph. Used during process of path deserialization
+// Supporting "only" <V, E> generic types implies that we should use enums to represent these types.
+impl<'a, V, E> FromSql<'a> for Path<V, E>
+where
+    V: Deserialize<'a>,
+    E: Deserialize<'a>,
+{
+    fn from_sql(
+        ty: &Type,
+        mut raw: &'a [u8],
+    ) -> Result<Path<V, E>, Box<dyn std::error::Error + Sync + Send>> {
+        if ty.schema() != "ag_catalog" || ty.name() != "agtype" {
+            return Err("Only ag_catalog.agtype is supported".into());
+        }
+
+        let mut b = [0; 1];
+        raw.read_exact(&mut b)?;
+
+        // We only support version 1 of the jsonb binary format
+        if b[0] != 1 {
+            return Err("unsupported JSONB encoding version".into());
+        }
+
+        if !(raw[0] == "[".as_bytes()[0] && &raw[raw.len() - 6..] == PATH_SUFFIX) {
+            return Err("Invalid path definition".into());
+        }
+
+        let mut vertices: Vec<Vertex<V>> = vec![];
+        let mut edges: Vec<Edge<E>> = vec![];
+
+        let mut first_open_bracket = raw.len();
+
+        for (i, b) in raw[..raw.len() - 7].iter().enumerate() {
```

Comment:
Change `(i, b)` to something more meaningful

Reply:
I kept `i` because it's used in so many places and after all it is just an index.